watertap/unit_models/tests/test_electroNP_ZO.py

```diff
@@ -0,0 +1,263 @@
+from idaes.core.util.scaling import (
+    calculate_scaling_factors,
+    unscaled_variables_generator,
+    unscaled_constraints_generator,
+    badly_scaled_var_generator,
+)
+from pyomo.util.check_units import assert_units_consistent
+from idaes.core import UnitModelCostingBlock
+from watertap.costing import WaterTAPCosting
+
+# -----------------------------------------------------------------------------
+# Get default solver for testing
+solver = get_solver()
+
+
+class TestElectroNP:
+    @pytest.fixture(scope="class")
+    def ElectroNP_frame(self):
+        m = ConcreteModel()
+        m.fs = FlowsheetBlock(dynamic=False)
+
+        m.fs.properties = ModifiedASM2dParameterBlock(
+            additional_solute_list=["S_K", "S_Mg"]
+        )
+
+        m.fs.unit = ElectroNPZO(property_package=m.fs.properties)
+
+        EPS = 1e-8
+
+        m.fs.unit.inlet.temperature.fix(298.15 * units.K)
+        m.fs.unit.inlet.pressure.fix(1 * units.atm)
+
+        m.fs.unit.inlet.flow_vol.fix(18446 * units.m**3 / units.day)
+        m.fs.unit.inlet.conc_mass_comp[0, "S_O2"].fix(10 * units.mg / units.liter)
+        m.fs.unit.inlet.conc_mass_comp[0, "S_N2"].fix(EPS * units.mg / units.liter)
+        m.fs.unit.inlet.conc_mass_comp[0, "S_NH4"].fix(16 * units.mg / units.liter)
+        m.fs.unit.inlet.conc_mass_comp[0, "S_NO3"].fix(EPS * units.mg / units.liter)
+        m.fs.unit.inlet.conc_mass_comp[0, "S_PO4"].fix(3.6 * units.mg / units.liter)
+        m.fs.unit.inlet.conc_mass_comp[0, "S_F"].fix(30 * units.mg / units.liter)
+        m.fs.unit.inlet.conc_mass_comp[0, "S_A"].fix(20 * units.mg / units.liter)
+        m.fs.unit.inlet.conc_mass_comp[0, "S_I"].fix(30 * units.mg / units.liter)
+        m.fs.unit.inlet.conc_mass_comp[0, "X_I"].fix(25 * units.mg / units.liter)
+        m.fs.unit.inlet.conc_mass_comp[0, "X_S"].fix(125 * units.mg / units.liter)
+        m.fs.unit.inlet.conc_mass_comp[0, "X_H"].fix(30 * units.mg / units.liter)
+        m.fs.unit.inlet.conc_mass_comp[0, "X_PAO"].fix(EPS * units.mg / units.liter)
+        m.fs.unit.inlet.conc_mass_comp[0, "X_PP"].fix(EPS * units.mg / units.liter)
+        m.fs.unit.inlet.conc_mass_comp[0, "X_PHA"].fix(EPS * units.mg / units.liter)
+        m.fs.unit.inlet.conc_mass_comp[0, "X_AUT"].fix(EPS * units.mg / units.liter)
+        m.fs.unit.inlet.conc_mass_comp[0, "X_MeOH"].fix(EPS * units.mg / units.liter)
+        m.fs.unit.inlet.conc_mass_comp[0, "X_MeP"].fix(EPS * units.mg / units.liter)
+        m.fs.unit.inlet.conc_mass_comp[0, "X_TSS"].fix(EPS * units.mg / units.liter)
+        m.fs.unit.inlet.conc_mass_comp[0, "S_K"].fix(EPS * units.mg / units.liter)
+        m.fs.unit.inlet.conc_mass_comp[0, "S_Mg"].fix(EPS * units.mg / units.liter)
+
+        # Alkalinity was givien in mg/L based on C
+        m.fs.unit.inlet.alkalinity[0].fix(61 / 12 * units.mmol / units.liter)
+
+        # Unit option
+        m.fs.unit.energy_electric_flow_mass.fix(0.044 * units.kWh / units.kg)
```

Comment:
Something just came to mind- we can probably later convert this model to not only focus on ZO formulation. Instead, we could consider having config options would would select between fixed performance or predictive performance from surrogates. In ZO config, removal fractions, etc., would just be fixed to variables, while in predictive mode, the constraint would be activated and the surrogate would be inserted there. Just food for thought.

Reply:
It is a good idea, we could definitely upgrade this model to be a surrogate modle with config option to choose, and I believe I can save a lot of time for building some basic vairables. I still name it as electroNP_ZO for now but would consider to rename it when we add the surrogate model.